router/src/server.rs

```diff
@@ -775,6 +776,10 @@ async fn chat_completions(
     let logprobs = logprobs.unwrap_or(false);
     let tool_prompt = tool_prompt.unwrap_or_default();
     let stop = stop.unwrap_or_default();
+    // rescale temperature starting from 0.0 to 1.0
+    let adjusted_temperature = temperature.map_or(1.0, |t| t + 1.0);
```

Comment:
I do not understand this. Why are you adding 1.0 ?

Reply:
1 is added since openais api is essentially zero indexed. They treat `0` as deterministic and higher values as more random. 

In our case 1 is deterministic and greater is more random, so adding 1 allows users to get deterministic output with temp 0, and more random with larger values.

> temperature:
> What sampling temperature to use, between 0 and 2. Higher values like 0.8 will make the output more random, while lower values like 0.2 will make it more focused and deterministic.

Reply:
No it's not.

logits.pow(temperature).


temperature == 0 -> Pretty much forces deteministic sampling (with float issues).
temperature == 1 -> Regular distribution.

Reply:
makes sense! PR updated to:
1. only set `do_sample` to `false` when `temperature` is 0
2. rescale `temperature` from 0 to 1 only when temp is 0

I believe this fits the requirements, and enables greedy ONLY when temperature is 0. Please lmk if I should change! 🙏